deploy/tests/network.ts

```diff
@@ -77,76 +81,320 @@ describe('network', () => {
+                await getNamedAccounts();
+
+            await expectRoleMembers(
+                bntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_SUPERVISOR,
+                [isMainnetFork() ? foundationMultisig : deployer]
+            );
+            await expectRoleMembers(
+                bntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_GOVERNOR,
+                isMainnet() ? [deployerV2] : [deployer]
+            );
+            await expectRoleMembers(
+                bntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_MINTER,
+                isMainnet()
+                    ? [standardRewards.address, bntPool.address, liquidityProtection, legacyStakingRewards]
+                    : [standardRewards.address, bntPool.address]
+            );
+
+            await expectRoleMembers(
+                vbntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_SUPERVISOR,
+                [isMainnetFork() ? foundationMultisig : deployer]
+            );
+            await expectRoleMembers(
+                vbntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_GOVERNOR,
+                isMainnet() ? [deployerV2] : [deployer]
+            );
+            await expectRoleMembers(
+                vbntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_MINTER,
+                isMainnet() ? [bntPool.address, liquidityProtection] : [bntPool.address]
+            );
+
+            await expectRoleMembers(masterVault, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig, network.address]);
+            await expectRoleMembers(masterVault, Roles.Vault.ROLE_ASSET_MANAGER, [
+                network.address,
+                poolCollection.address
+            ]);
+
+            await expectRoleMembers(externalProtectionVault, Roles.Upgradeable.ROLE_ADMIN, [
+                daoMultisig,
+                network.address
+            ]);
+            await expectRoleMembers(externalProtectionVault, Roles.Vault.ROLE_ASSET_MANAGER, [
+                network.address,
+                poolCollection.address
+            ]);
+
+            await expectRoleMembers(externalRewardsVault, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+            await expectRoleMembers(externalRewardsVault, Roles.Vault.ROLE_ASSET_MANAGER, [standardRewards.address]);
+
+            await expectRoleMembers(poolTokenFactory, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+
+            await expectRoleMembers(networkSettings, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+
+            await expectRoleMembers(bntPool, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig, network.address]);
+            await expectRoleMembers(bntPool, Roles.BNTPool.ROLE_BNT_POOL_TOKEN_MANAGER);
```

Comment:
will be the auto compounding rewards, right?

Reply:
Yeah (see here https://github.com/bancorprotocol/contracts-v3/pull/339/files#diff-51e7f998be6b9092c56b78c26d1d65efeb279ad11f2b6b7a16c6219d077f4082)

Reply:
cool 👍